subprojects/language-java/language-java.gradle.kts

```diff
@@ -22,6 +22,7 @@ dependencies {
     implementation(project(":core"))
     implementation(project(":workers"))
     implementation(project(":snapshots"))
+    implementation(project(":execution"))
```

Comment:
This doesn't look good.

Reply:
`RecompilationSpecProvider` uses `DefaultFileChange` to re-create the classpath changes for consumption of the incremental Java compilation. That can go away when we use input changes.

Maybe the whole alignment can go away now completely, since we report the class path changes differently. I would not couple the changes to this PR though.

Note that incremental Java compilation uses the `FileSystemSnapshotter` as well. The same holds true here: I wonder if we actually need to snapshot the classpath entries or if we can re-use what `InputChanges` provides. I think `InputChanges` should provide enough.

Reply:
Sounds good. I'd sleep better if we had some checks for unused `implementation` dependencies, so once we start using `InputChanges` we'd figure out that `:execution` is not needed anymore.